### ISSUE TITLE
batched_vector_arithmatic_simd_dcomplex{2,4} failures w/ GCC 10.2.0 & c++17, with or without blas tpl enabled

### ISSUE THREAD

**Issue by e10harvey** [kokkos/kokkos-kernels#1512]
Arithmetic failures w/ GCC 10.2.0 & c++17 in serial backend:
```
[ RUN      ] serial.batched_vector_arithmatic_simd_dcomplex2
/path/to/kokkos-kernels/unit_test/batched/dense/Test_Batched_VectorArithmatic.hpp:90: Failure
The difference between ats::abs(c[k]) and ats::abs(alpha + b[k]) is 1.1944432591914926, which exceeds eps * ats::abs(c[k]), where
ats::abs(c[k]) evaluates to 1.9116719714609276,
ats::abs(alpha + b[k]) evaluates to 0.71722871226943496, and
eps * ats::abs(c[k]) evaluates to 4.2447644764929739e-13.
<snip>
[ RUN      ] serial.batched_vector_arithmatic_simd_dcomplex4
/path/to/kokkos-kernels/unit_test/batched/dense/Test_Batched_VectorArithmatic.hpp:90: Failure
The difference between ats::abs(c[k]) and ats::abs(alpha + b[k]) is 0.58011900740885669, which exceeds eps * ats::abs(c[k]), where
ats::abs(c[k]) evaluates to 1.9116719714609276,
ats::abs(alpha + b[k]) evaluates to 1.3315529640520709, and
eps * ats::abs(c[k]) evaluates to 4.2447644764929739e-13.
<snip>
```
Similar arithmetic failures occur in the threads and openmp backends.

### Reproducer
```
source /etc/profile.d/modules.sh
module purge
module load cmake/3.19.3 gcc/10.2.0

$KOKKOSKERNELS_PATH/cm_generate_makefile.bash --with-devices=Threads,Serial --arch=SKX --compiler=$CXX --cxxflags="-O3 -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized " --cxxstandard="17" --ldflags=""   --kokkos-path=$KOKKOS_PATH --kokkoskernels-path=$KOKKOSKERNELS_PATH --with-scalars='double,complex_double' --with-ordinals=int --with-offsets=int,size_t --with-layouts=LayoutLeft --with-tpls=    --with-options= --with-cuda-options=   --no-examples
```

Note that `batched_vector_arithmatic_simd_dcomplex3` passes in the serial, openmp, and threads backend.

**Comment by ndellingwood** [comment]
Cross-referencing #969 (I think same test failure with gcc/10.2.0)

**Comment by ndellingwood** [comment]
Can this be closed with merged PR #1520?
